src/services/product.service.ts

```diff
@@ -30,32 +42,94 @@ export interface ProductDetailsResponse {
   rating: number
 }
 
-export interface ProductFilters {
-  name: string;
-  provider: string;
-  brand: string;
+export interface ProductCriteria {
+    phrase?: string;
+    ingredientsToIncludeIds?: string[];
+    ingredientsToExcludeIds?: string[];
+    minRating?: number;
+    // TODO: sort by, brands, providers, categories
 }
 
-export const getProductsListApi = (): Promise<
-  AxiosResponse<ProductResponse>
-> => axios.get(`${productsApiUrl}`);
+export const urlToProductCriteria = (url: string): ProductCriteria => {
+  const queryParams = new URLSearchParams(url);
+  const minRatingStr = queryParams.get(ProductListRequestParams.MIN_RATING);
+  let minRating: number | undefined = minRatingStr !== null
+    ? parseInt(minRatingStr, 10) : undefined;
 
-export const getFilteredProductsListApi = (
-  params: ProductFilters,
-): Promise<AxiosResponse<ProductResponse>> => {
-  const filteredParams = Object.fromEntries(
-    Object.entries(params).filter(([_, value]) => value !== ''),
-  );
+  if (Number.isNaN(minRating as number)) {
```

Comment:
why not check if it is undefined here as it can only be number or undefined by your definition? i do not like the idea of casting undefined to a number